src/api/app/controllers/webui/distributions_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Webui::DistributionsController < Webui::WebuiController
+  before_action :set_project
+  before_action :set_distribution, only: [:toggle]
+  after_action :verify_authorized
+
+  # GET /projects/:project/distributions/new
+  def new
+    authorize @project, :update?
+
+    @distributions = Distribution.all.group_by(&:vendor)
+    return if @distributions.present?
+
+    if User.admin_session?
```

Comment:
couldn't you instead of write custom code, use the `before_action :require_admin` ?

Reply:
I'm not writing custom code, I'm copy and pasting it 😄  `require_admin` wouldn't redirect to the interconnect.